### PR TITLE
Task s4 db queries

### PR DESCRIPTION
**Pull request by C0deJack**
Completed task of creating a DB connection for the single product page. Also sorted merge conflicts from master (story2). File is missing the class of Product that is not in this pull request but in story 3

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -19,26 +19,31 @@
...
 <body>
-    <div class="row">
-        <nav class="navbar-fixed-top col-xs-9 col-sm-10 col-xs-offset-3 col-sm-offset-2">
-            <h1>Totally Pawesome</h1>
-        </nav>
-    </div>
-    <aside class="col-xs-3 col-sm-2">
+<section class="row banner">
+    <div class="home col-xs-4 col-sm-2">
         <a href="#">
```
Comment (kyam-mayden) [inline index.php]: Please make this direct to index page

File: src/app/productPage.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+namespace store;
```
Comment (kyam-mayden) [inline src/app/productPage.php]: store should be 'Store'

File: src/app/productPage.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+require_once '../../vendor/autoload.php';
+
+use Store\mySqlDbConnect as mySqlDbConnect;
+use Store\Store as Store;
+
+$mySqlCon = new mySqlDbConnect();
+$store = new Store($mySqlCon);
+$product = $store->getProductPage(1);
+
+var_dump($product);
```
Comment (kyam-mayden) [inline src/app/productPage.php]: please remove var dump

File: src/assets/css/styles.css [hunk 4]
```diff
@@ -1,26 +1,48 @@
...
-    height: 100px;
+.row {
+    margin: 0;
 }
 
-aside {
-    height: 100vh;
-    background-color: #708090;
+.banner {
+    color: #ffffff;
```
Comment (marty-crane) [inline src/assets/css/styles.css]: this is being achieved on line 19

File: src/assets/css/styles.css [hunk 5]
```diff
@@ -1,26 +1,48 @@
...
 }
 
-aside a {
+a:hover {
+    text-decoration: none;
+}
+
+.banner a, .sidebar a {
```
Comment (kyam-mayden) [inline src/assets/css/styles.css]: if you change this to .banner, .sidebar a{ you can remove line 10/11

File: src/classes/Store.php [hunk 6]
```diff
@@ -1,26 +1,50 @@
...
     }
 
-    public static function setPDO(PDO $db){
-        self::$DB = $db;
+    /**
+     *Database query to get product data
+     *
+     * @return  Product class object
+     */
+
```
Comment (kyam-mayden) [inline src/classes/Store.php]: line space

File: src/classes/Store.php [hunk 7]
```diff
@@ -1,26 +1,50 @@
...
+        $query = $this->db->prepare("SELECT `id`, `categoryName`, `defaultImageFilePath`,`defaultImageAlt` FROM `categories` WHERE `deleted` = 0;");
         $query->execute();
         return $query->fetchAll(PDO::FETCH_CLASS, Category::class);
     }
 
-    public static function setPDO(PDO $db){
-        self::$DB = $db;
+    /**
+     *Database query to get product data
+     *
```
Comment (kyam-mayden) [inline src/classes/Store.php]: this takes a param

File: src/classes/Store.php [hunk 8]
```diff
@@ -1,26 +1,50 @@
...
+     * @return  Product class object
+     */
+
+
+    public function getProductPage($id): Product {
```
Comment (kyam-mayden) [inline src/classes/Store.php]: input type hinting needed

File: src/classes/Store.php [hunk 9]
```diff
@@ -1,26 +1,50 @@
...
+        $query2 = $this->db->prepare("SELECT `products`.`id`,
```
Comment (kyam-mayden) [inline src/classes/Store.php]: why is this not $query as opposed to $query2?

File: src/classes/interfaces/GetProductPageable.php [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Store\interfaces;
```
Comment (kyam-mayden) [inline src/classes/interfaces/GetProductPageable.php]: Should be capital 'I'

File: src/classes/interfaces/connectToDb.php [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Store\interfaces;
```
Comment (kyam-mayden) [inline src/classes/interfaces/connectToDb.php]: See above comment

File: src/classes/mySqlDbConnect.php [hunk 12]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Store;
+
+use \PDO;
+use \Exception;
+use Store\interfaces\ConnectToDb;
```
Comment (kyam-mayden) [inline src/classes/mySqlDbConnect.php]: See above comment

**Review by kyam-mayden — CHANGES_REQUESTED** [review]
Some very minor changes to look at

File: src/assets/css/styles.css [hunk 13]
```diff
@@ -7,15 +7,11 @@ html, body {
...
-.banner {
-    color: #ffffff;
-}
-
 a:hover {
     text-decoration: none;
 }
 
-.banner a, .sidebar a {
+.banner a, .banner, .sidebar a {
```
Comment (kyam-mayden) [inline src/assets/css/styles.css:13]: As text color is inherited you can probably lose the '.banner a'

**Review by kyam-mayden — APPROVED** [review]
Love it

**Pull request merged**